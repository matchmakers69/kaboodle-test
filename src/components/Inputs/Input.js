import React from 'react';
import styles from './Styles.module.scss';

const Input = ({ name, type, value, maxLength, placeholder, onChange }) => {
  return (
    <input
      onChange={onChange}
      placeholder={placeholder}
      maxLength={maxLength}
      value={value}
      type={type}
      name={name}
    />
  );
};

export default Input;
