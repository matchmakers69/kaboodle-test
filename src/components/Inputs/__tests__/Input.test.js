import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from '../Input';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const props = {
  onChange: jest.fn(),
  placeholder: 'Search via postcode eg. 25',
  maxLength: '6',
  value: '199',
  type: '',
  name: ''
};

describe('<Input />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Input {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  // it('Input component exists', () => {
  //   const component = findByTestAttr(Input, 'accomodationInput');
  //   expect(component.length).toBe(1);
  // });

  // it('Input should have placeholder', () => {
  //   expect(
  //     Input.find('[data-test="accomodationInput"]').props().placeholder
  //   ).toBe('Search via postcode eg. 25');
  // });

  // it('renders input with given value', () => {
  //   expect(Input.find('[data-test="accomodationInput"]').props().value).toEqual(
  //     '198'
  //   );
  // });

  // it('simulate change', () => {
  //   const accomodationPostCode = updateInput(
  //     Input,
  //     '[data-test="accomodationInput"]',
  //     '198',
  //     '198'
  //   );
  //   expect(props.onChange).toBeCalledTimes(1);
  //   Input.find('[data-test="accomodationInput"]').simulate('change');

  //   expect(accomodationPostCode.props().value).toBe('198');
  // });
});
