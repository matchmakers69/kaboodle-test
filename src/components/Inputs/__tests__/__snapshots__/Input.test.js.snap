// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
<input
  className="accomodationInput"
  data-test="accomodationInput"
  maxLength="6"
  name=""
  onChange={[MockFunction]}
  placeholder="Search via postcode eg. 25"
  type=""
  value="199"
/>
`;
